### ISSUE TITLE
Return type of property setters is not checked

### ISSUE THREAD

**Issue by decorator-factory** [python/mypy#20513]
Example code:

```py
class Banana:
    @property
    def size(self) -> int:
        return 42

    @size.setter
    async def size(self, value: int, /) -> None:
        print(f"set value to {value!r}")

b = Banana()
b.size = 42
```

I expected mypy to complain that the setter function doesn't return `None` and returns `Coroutine[Any, Any, None]` instead. Similarly, if the setter is a non-async function returning e.g. `str` or `int`, no error is reported.

### Environment
- Mypy version used: 1.19.1
- Mypy command-line flags: `--strict`
- Python version used: 3.14

**Comment by TeamSpen210** [comment]
Is this actually a typing error? `property` [discards](https://github.com/python/cpython/blob/61fc72a4a431cbfd42f22e2af76177c73431c3e6/Objects/descrobject.c#L1747-L1760) the return value, so any return value is okay. Definitely not a desirable result though. Maybe there should be a diagnostic emitted when coercing any `Awaitable`-returning callable to one returning `object`, since code expecting a sync callable won't usually handle async ones.

**Comment by decorator-factory** [comment]
Various magic methods, such as `__del__`, `__init_subclass__`, `__setattr__`, `__delattr__`, do ignore the return value and skip checking if it's `None` (at this point it would be a big backwards incompatible change). However, returning something else is probably a mistake, and `mypy` already does report an error for `__setattr__`, `__delattr__` and `__init_subclass__` returning not-`None` values. 

As a data point, `ty` does flag this code as an error, but `pyright` and `pyrefly` don't

P.S. I wonder why `__init__` does check the return value but not `__init_subclass__`?
